### PR TITLE
FE-960: [adapter] Add get ETA API to the adapter

### PR DESCRIPTION
**Pull request by adubovsky**
Also contains important fix for `ajax` util around auth header handling.

@Glympse/web, PTAL.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 version |date        |notes
 :-------|:-----------|:------
+2.8.x   | 2017/07/?? | BugFix for auth requests. New `core.getETAInfo` endpoint + `ETAInfo` event.
```
Comment (j5kay) [inline CHANGELOG.md]: This will be a 2.9.x release.. new features

File: app/src/GlympseAdapterDefines.js [hunk 2]
```diff
@@ -95,6 +95,7 @@ define(function(require, exports, module)
 				, getGroups: 'getGroups'
 				, removeGroup: 'removeGroup'
 				, getOrgObjects: 'getOrgObjects'
+				, getETAInfo: 'getETAInfo'
```
Comment (j5kay) [inline app/src/GlympseAdapterDefines.js]: Please keep casing consistent -- `getEtaInfo`

File: app/src/GlympseAdapterDefines.js [hunk 3]
```diff
@@ -153,6 +154,8 @@ define(function(require, exports, module)
 			, GroupStatus: 'GroupStatus'
 			, OrgObjects: 'OrgObjects'
 
+			, ETAInfo: 'ETAInfo'
```
Comment (j5kay) [inline app/src/GlympseAdapterDefines.js]: Should be `EtaInfo`

File: app/src/adapter/CoreController.js [hunk 4]
```diff
@@ -174,6 +183,72 @@ define(function(require, exports, module)
...
 
+		/**
+		 * Fetch ETA information to the given points
+		 *
+		 * @param {Object[]} routes - array of routes for which to calculate ETAs.
+		 *	Each object must has the following properties:
+		 *	- "start" - is a starting point in format { lat: number, lng: number }
+		 *	- "end" - is an ending point in format { lat: number, lng: number }
+		 */
+		function getETAInfo(routes)
```
Comment (j5kay) [inline app/src/adapter/CoreController.js]: `getEtaInfo`

File: app/src/test-client/src/Main.js [hunk 5]
```diff
@@ -43,6 +43,22 @@ define(function(require, exports, module)
 					dbg('--> ADAPTER READY', args);
 					invitesCard = args.cards;
 					invitesGlympse = args.glympses;
+
+					// adapter.core.getETAInfo([
```
Comment (j5kay) [inline app/src/test-client/src/Main.js]: Still needed?

**Comment by j5kay** [comment]
LGTM

**Comment by adubovsky** [comment]
All fixed, merging...

**Pull request merged**